tests/functional/WebDriverTestCase.php

```diff
@@ -46,8 +47,12 @@ protected function setUp(): void
 
             if ($browserName === WebDriverBrowserType::CHROME) {
                 $chromeOptions = new ChromeOptions();
+
+                if (!$disableHeadless) {
```

Comment:
There is also similar case for starting a headless Firefox few lines below - this should be handled as well.

Also just a personal preference, but I would prefer moving this condition **below** the already existing `$chromeOptions->addArguments(`... call.